### PR TITLE
Pixel shifting for ERP

### PR DESCRIPTION
**Pull request by monster860**
This code is completely 100% untested.

## About The Pull Request

What this does is make it so that if you are holding ctrl+shift and then do a direction it causes you to pixel shift

## Why It's Good For The Game

well you see I was doing a threesome ERP and I was on top of someone with someone else and we couldn't position ourselves correctly so I open this PR so that no one else has to deal with the limitations of tile based movement when trying to ERP.

## Changelog
:cl:
add: Use Ctrl-Shift-direction key to shift your characters position. Use for ERP.
/:cl:

### CODE REVIEW

**Comment by monster860** [comment]
Against my better judgement I tested the PR. Here's a picture

image

**Comment by monster860** [comment]
image

**Comment by BlackMajor** [comment]
PR of the year.
Make sure it works for other mobs too. Like ash drakes, borgs and xenos.

**Comment by monster860** [comment]
it should work on all mobs

**Comment by BlackMajor** [comment]
Bless.

**Comment by dapnee** [comment]
absolute madlad. 
now for this to be used for combat /s

**Comment by Ragolution** [comment]
> absolute madlad.
> now for this to be used for combat /s

If it doesn't already do so on tile transition, this should have a line specifying a recenter on movement, to prevent abuse.

**Comment by Detective-Google** [comment]
> > absolute madlad.
> > now for this to be used for combat /s
> 
> If it doesn't already do so on tile transition, this should have a line specifying a recenter on movement, to prevent abuse.

Should also be a limit to prevent shifting out of the tile they're currently in(don't want people shifting their sprites into walls and whatnot.)

**Comment by monster860** [comment]
you can't shift more than 16 pixels in a direction to prevent aboos

**Review by Ragolution — APPROVED** [review]
I think it's funny I can even approve things.

**Comment by Ghommie** [comment]
The pixel shift doesn't reset on move, perhaps add two variables to store the x and y shifted pixels amount and a comsig hook to reset them on move.

**Review by Ghommie — CHANGES_REQUESTED** [review]
The pixel shifting doesn't reset on move, perhaps add two variables to store the x and y shifted pixels amount and a comsig hook to do the job.

**Comment by monster860** [comment]
there I did the thing

File: code/modules/mob/mob_defines.dm [hunk 1]
```diff
@@ -38,6 +38,7 @@
 	var/resting = 0			//Carbon
 	var/lying = 0
 	var/lying_prev = 0
+	var/is_shifted = 0
```
Comment (silicons) [inline code/modules/mob/mob_defines.dm]:
```suggestion
	var/is_shifted = FALSE
```

**Comment by Detective-Google** [comment]
~~@monster860 as of commit c8535275ce being testmerged on the server, using the "set custom holoform" pAI option and then swapping your holo to the new one whilst pixel shifted will result in your "default" position to being the position you were shifted in when you set the holoform.~~

Apparently it's an issue with tauric bodies(specifically the naga) instead? unsure from my testing.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**